src/screens/Forum/ForumPost.js

```diff
@@ -1,62 +1,126 @@
-import React from 'react';
+import React, { useState } from 'react';
 import {
   Text, TouchableOpacity, StyleSheet,
 } from 'react-native';
-import { Card, Button, Title } from 'react-native-paper';
+import {
+  Card, Title, ActivityIndicator, IconButton, Menu, Divider,
+} from 'react-native-paper';
+import firestore from '@react-native-firebase/firestore';
 import PropTypes from 'prop-types';
+import { theme } from '../../style';
 
 const styles = StyleSheet.create({
-  container: {
+  postContainer: {
     marginTop: 10,
     width: '95%',
     alignSelf: 'center',
     borderRadius: 15,
   },
-  title: {
-  },
   actionContainer: {
     width: '100%',
     marginHorizontal: '1%',
     justifyContent: 'space-between',
   },
-  text: {
+  bodyText: {
+    color: theme.colors.text,
+    fontFamily: theme.fonts.medium.fontFamily,
+    fontWeight: theme.fonts.medium.fontWeight,
+  },
+  sideText: {
+    color: theme.colors.accent,
+    fontFamily: theme.fonts.regular.fontFamily,
+    fontWeight: theme.fonts.regular.fontWeight,
   },
 });
 
 export default function ForumPost({
-  name, time, numReplies, children,
+  userID, time, children, postID, navigateToPostScreen, belongsToCurrentUser,
 }) {
+  const [loading, setLoading] = useState(true);
+  const [numReplies, setNumReplies] = useState(0);
+
+  firestore().collection('forum_comments').where('postID', '==', postID)
+    .get()
+    .then((querySnapshot) => {
+      setNumReplies(querySnapshot.size);
+      setLoading(false);
+    });
+
+  const [name, setName] = useState('Name unset');
+  const [isAdmin, setIsAdmin] = useState(false);
+  const [userErrorMessage, setUserErrorMessage] = useState();
+
+  /* Get user w/ this userID from database, the user's display name, and if they are PTV staff */
+  /* ERROR isn't getting user information */
```

Comment:
I think you're getting the error because some of the `userID`s you're looking for don't exist in the `users` table. We should clean up the database.